perfkitbenchmarker/benchmark_spec.py

```diff
@@ -228,6 +233,34 @@ def ConstructVirtualMachines(self):
       self.vm_groups[group_name] = vms
       self.vms.extend(vms)
 
+  def ConstructSparkService(self):
+    if self.config.spark_service is None:
+      return
+
+    providers.LoadProvider(self.config.spark_service.cloud)
+    spark_service_spec = self.config.spark_service
+    service_type = spark_service_spec.spark_service_type
+    if service_type == spark_service.PKB_MANAGED:
+      spark_service_class = spark_service.GetSparkService(
+          spark_service.PKB_MANAGED)
+    elif service_type == spark_service.PROVIDER_MANAGED:
+      spark_service_class = spark_service.GetSparkService(
+          spark_service_spec.cloud)
+    else:
+      valid_types = ' '.join(spark_service._SPARK_SERVICE_REGISTRY.keys())
+      raise Exception('Invalid spark service type %s must be one of %s',
+                      spark_service_spec.spark_service_type, valid_types)
+    if spark_service_class is None:
```

Comment:
It looks like you already raise an error in GetSparkService if it couldn't find a class corresponding to the cloud